### PR TITLE
Remove nginx default profile

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind deprecation

#### What this PR does / why we need it:
The profile is outdated and we have no automation in place to update it. Means we remove it for now per:
https://github.com/kubernetes-sigs/security-profiles-operator/issues/1359

#### Which issue(s) this PR fixes:

None
#### Does this PR have test?
None

#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
Removed default nginx profile from operator deployment.
```

### CODE REVIEW

File: test/tc_run_pod_test.go [hunk 1]
```diff
@@ -76,12 +76,4 @@ func (e *e2e) testCaseRunPod([]string) {
 
 	e.logf("Waiting for test pod to be ready")
 	e.waitFor("condition=ready", "pod", "--all")
-
```
Comment (jhrozek) [inline test/tc_run_pod_test.go:79]: I wonder if the test case as a whole has much of value right now?
If we want to keep a negative test around maybe we could add the failure case:
```
	failureOutput := e.kubectlFailure(
		"exec", examplePodName, "--", "rmdir", "/home",
	)
	e.Contains(failureOutput,
		"rmdir: failed to remove '/home': Operation not permitted",
	)
```
to `testCaseBaseProfile` ?

Comment (saschagrunert) [reply test/tc_run_pod_test.go:79]: Yeah I agree and decided to remove the whole test case for now.

**Review by ccojocar — APPROVED** [review]
I am looking forward to see this merged. Thanks for removing it.

**Comment by ccojocar** [comment]
@saschagrunert is there any chance to get this merged soon? The default nginx profile is causing some issues on our side. I cannot manually remove it because the operator keeps install it back. Thanks

**Comment by saschagrunert** [comment]
Rebased on top of the latest `main` branch.

**Pull request merged**